src/dataflow/sink/kafka.rs

```diff
@@ -22,26 +22,26 @@ use timely::dataflow::{Scope, Stream};
 
 use dataflow_types::{Diff, KafkaSinkConnector, Timestamp};
 use expr::GlobalId;
-use interchange::avro::Encoder;
+use interchange::avro::{DiffPair, Encoder};
 use ore::collections::CollectionExt;
 use repr::{RelationDesc, Row};
 
-// TODO@jldlaughlin: What guarantess does this sink support? #1728
+// TODO@jldlaughlin: What guarantees does this sink support? #1728
 pub fn kafka<G>(
     stream: &Stream<G, (Row, Timestamp, Diff)>,
     id: GlobalId,
     connector: KafkaSinkConnector,
-    relation_desc: RelationDesc,
+    desc: RelationDesc,
 ) where
     G: Scope<Timestamp = Timestamp>,
 {
-    let schema = interchange::avro::encode_schema(&relation_desc).expect("");
-
+    let encoder = Encoder::new(desc);
```

Comment:
this line messed me up for a bit (converting the immutable RelationDesc to a mutable RelationDesc) so I just want to check my understanding

when we call this function we move desc into the scope of Encoder::new -- this function can't do anything with it anymore (its not a copy because this type doesn't implement Copy) and you are free to move a immutable variable into a mutable variable because .... I guess its the owning function's right to do whatever it wants?